src/Sylius/Behat/Context/Ui/Shop/ProductContext.php

```diff
@@ -525,6 +532,51 @@ public function iShouldBeInformedThatTheProductDoesNotExist()
         Assert::eq($this->errorPage->getTitle(), 'The "product" has not been found');
     }
 
+    /**
+     * @When /^I try to browse products from (taxon "([^"]+)")$/
+     */
+    public function iTryToBrowseProductsFrom(TaxonInterface $taxon)
+    {
+        $this->indexPage->tryToOpen(['slug' => $taxon->getSlug()]);
+    }
+
+    /**
+     * @Then I should be informed that the taxon does not exist
+     */
+    public function iShouldBeInformedThatTheTaxonDoesNotExist()
+    {
+        Assert::eq($this->errorPage->getTitle(), 'Requested page is invalid.');
```

Comment:
`eq` -> `same`

Reply:
Also, cannot we get some more explainable error message from the page?

Reply:
I don't really know. Taxons and products use the same controller: taxons use indexAction and products use showAction. The indexAction of the resource controller does not call the internal method findOr404, so we can't  easily get a 404 so that the page will print "The taxon has not been found", as it does for products.